components/operations-controller/internal/webhook/client.go

```diff
@@ -207,7 +221,13 @@ func parseResponseObject(resp *http.Response) (*web_hook.ResponseObject, error)
 func checkForErr(resp *http.Response, successStatusCode *int, error *string) error {
 	var errMsg string
 	if *successStatusCode != resp.StatusCode {
-		errMsg += fmt.Sprintf("response success status code was not met - expected %q, got %q; ", *successStatusCode, resp.StatusCode)
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to parse webhook HTTP response with body %s", respBody))
+		}
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(respBody))
+
+		errMsg += fmt.Sprintf("response success status code was not met - expected %d, got %d and body %s; ", *successStatusCode, resp.StatusCode, respBody)
```

Comment:
have you tried out how such an error would look like ? To me it looks like if an error is present in the template, and you also add the body, you will get the same error twice in the message..

Reply:
Yes, you're absolutely correct. It's my fault here because I had setup a mock server which didn't return an error JSON in the response body, thus the `error` field in the output template was empty. That's why I didn't see any good error messages in the first place and that's why I though that it would be a good idea to inject the response body in there. I'll correct this.